ports/esp8266/esppwm.c

```diff
@@ -222,14 +231,13 @@ pwm_set_duty(int16_t duty, uint8 channel) {
         return;
     }
 
-    LOCK_PWM(critical);   // enter critical
     if (duty < 1) {
-        pwm.duty[channel] = 0;
+        duty = 0;
     } else if (duty >= PWM_DEPTH) {
-        pwm.duty[channel] = PWM_DEPTH;
-    } else {
-        pwm.duty[channel] = duty;
+        duty = PWM_DEPTH;
     }
+    LOCK_PWM(critical);   // enter critical
+    pwm.duty[channel] = duty;
```

Comment:
Did you test that this change to this function was needed?  As far as I can tell the `critical` lock in this file doesn't really do anything because it's only protecting the top-level API from itself, it doesn't protect against the `pwm_tim1_intr_handler`.

Reply:
Sure, but since I was reworking the file, I found this change harmless.

Reply:
It's best not to make unneeded changes, especially for something which fixes subtle bugs as done here.  When someone goes to review the commit (in the future maybe because it introduced a bug) then it's much better to have a simple-as-possible commit that fixes the problem.

If you think this change is important and worth including then please put it in a separate commit with a reason/justification.

Reply:
ok, removed